Mlem/Views/Tabs/Feeds/FeedView.swift

```diff
@@ -0,0 +1,225 @@
+    @Environment(\.scrollViewProxy) private var scrollViewProxy
+    @Environment(\.horizontalSizeClass) private var horizontalSizeClass
+    @EnvironmentObject var appState: AppState
+
+    @State var feedType: FeedType
+    
+    // MARK: Feed
+    
+    @StateObject var postTracker: PostTracker
+    @State var postSortType: PostSortType
+    
+    @Binding var rootDetails: CommunityLinkWithContext?
+    @Binding var splitViewColumnVisibility: NavigationSplitViewVisibility
+    
+    // MARK: Scroll to top
+    
+    @Namespace var scrollToTop
+    @State private var scrollToTopAppeared = false
+    private var scrollToTopId: Int? {
+        postTracker.items.first?.id
+    }
+    
+    // MARK: Destructive confirmation
+    
+    @State private var isPresentingConfirmDestructive: Bool = false
+    @State private var confirmationMenuFunction: StandardMenuFunction?
+    
+    func confirmDestructive(destructiveFunction: StandardMenuFunction) {
+        confirmationMenuFunction = destructiveFunction
+        isPresentingConfirmDestructive = true
+    }
+    
+    init(
+        feedType: FeedType,
+        splitViewColumnVisibility: Binding<NavigationSplitViewVisibility>? = nil,
+        rootDetails: Binding<CommunityLinkWithContext?>? = nil
+    ) {
+        // need to grab some stuff from app storage to initialize post tracker with
+        @AppStorage("internetSpeed") var internetSpeed: InternetSpeed = .fast
+        @AppStorage("upvoteOnSave") var upvoteOnSave = false
+        
+        self._feedType = State(initialValue: feedType)
+        
+        self._splitViewColumnVisibility = splitViewColumnVisibility ?? .constant(.automatic)
+        self._rootDetails = rootDetails ?? .constant(nil)
+        
+        @AppStorage("defaultPostSorting") var defaultPostSorting: PostSortType = .hot
+        self._postSortType = .init(wrappedValue: defaultPostSorting)
+        
+        self._postTracker = StateObject(wrappedValue: .init(
+            shouldPerformMergeSorting: false,
+            internetSpeed: internetSpeed,
+            upvoteOnSave: upvoteOnSave,
+            type: .feed(feedType, sortedBy: defaultPostSorting)
+        ))
+    }
+    
+    var body: some View {
+        ScrollView {
+            VStack(spacing: 0) {
+                VStack(spacing: 0) {
+                    ScrollToView(appeared: $scrollToTopAppeared)
+                        .id(scrollToTop)
+                    headerView
+                        .padding(.top, 5)
+                }
+                .background(Color.systemBackground)
+                if !postTracker.items.isEmpty {
+                    Divider()
+                }
+                PostFeedView(postTracker: postTracker, postSortType: $postSortType)
+                    .background(Color.secondarySystemBackground)
+            }
+        }
+        .refreshable {
+            await postTracker.refresh(clearBeforeFetch: true)
+        }
```

Comment:
Should be wrapped in a `Task` to prevent view redraw cancellation problems:

```suggestion
        .refreshable {
            await Task {
                _ = await postTracker.refresh(clearBeforeFetch: true)
            }.value
        }
```

Reply:
Also not sure how I feel about clearBeforeFetch being on here, it causes a weird little flicker if the refresh is fast that isn't super great

Reply:
Oops, I didn't mean to put that here haha. Nice catch.